core/src/main/java/com/cloud/storage/template/HttpTemplateDownloader.java

```diff
@@ -218,7 +219,10 @@ public long download(boolean resume, DownloadCompleteCallback callback) {
             errorString = hte.getMessage();
         } catch (IOException ioe) {
             status = TemplateDownloader.Status.UNRECOVERABLE_ERROR; //probably a file write error?
-            errorString = ioe.getMessage();
+            // Let's not overwrite the original error message.
+            if (errorString == null){
```

Comment:
Minor format one: can you leave a space between the closing parenthesis and the opening brace?